api/v1alpha1/kyma_types.go

```diff
@@ -273,17 +287,16 @@ type moduleStatusExistsPair struct {
 }
 
 func (kyma *Kyma) GetNoLongerExistingModuleStatus() []*ModuleStatus {
-	moduleStatusMap := make(map[string]*moduleStatusExistsPair)
+	moduleStatusMap := make(map[int]*moduleStatusExistsPair)
 
 	for i := range kyma.Status.ModuleStatus {
 		moduleStatus := &kyma.Status.ModuleStatus[i]
-		moduleStatusMap[moduleStatus.ModuleName] = &moduleStatusExistsPair{exists: false, moduleStatus: moduleStatus}
+		moduleStatusMap[i] = &moduleStatusExistsPair{exists: false, moduleStatus: moduleStatus}
```

Comment:
Use index here will not work I assume, we can't keep the `ModuleStatus` and `Modules` same order.

Reply:
Why can we not keep them in the same order?

Reply:
changed to array indexing